File: flutter_mort/lib/data/models/trust_safety.dart
class IdentityVerificationStatus {
  const IdentityVerificationStatus({
    required this.status,
    required this.verificationLevel,
    required this.marketplaceEnabled,
    required this.guardianModeOptional,
    required this.verificationMode,
    required this.productionVerified,
    required this.sandboxEligible,
    required this.testMode,
    required this.submissionsEnabled,
    required this.productionProviderAvailable,
    required this.canRetry,
    required this.supportEscalationAvailable,
    required this.documentsCollectedByMort,
    this.id,
    this.role,
    this.evidenceRoute,
    this.environment,
    this.provider,
    this.providerReference,
    this.decisionSource,
    this.expiresAt,
    this.rejectionCode,
    this.publicMessage,
    this.failureCode,
    this.privacyNoticeVersion,
    this.evidenceTypes = const [],
  });

  factory IdentityVerificationStatus.fromMap(Map<String, dynamic> map) {
    return IdentityVerificationStatus(
      id: map['id'] as String?,
      role: map['role'] as String?,
      status: map['status'] as String? ?? 'unverified',
      evidenceRoute: map['evidence_route'] as String?,
      verificationLevel: (map['verification_level'] as num?)?.toInt() ?? 0,
      marketplaceEnabled: map['marketplace_enabled'] == true,
      guardianModeOptional: map['guardian_mode_optional'] != false,
      verificationMode: map['verification_mode'] as String? ?? 'disabled',
      productionVerified: map['production_verified'] == true,
      sandboxEligible: map['sandbox_eligible'] == true,
      testMode: map['test_mode'] == true,
      submissionsEnabled: map['submissions_enabled'] == true,
      productionProviderAvailable: map['production_provider_available'] == true,
      canRetry: map['can_retry'] == true,
      supportEscalationAvailable: map['support_escalation_available'] != false,
      documentsCollectedByMort: map['documents_collected_by_mort'] == true,
      environment: map['environment'] as String?,
      provider: map['provider'] as String?,
      providerReference: map['provider_reference'] as String?,
      decisionSource: map['decision_source'] as String?,
      expiresAt: map['expires_at'] as String?,
      rejectionCode: map['rejection_code'] as String?,
      publicMessage: map['public_message'] as String?,
      failureCode: map['failure_code'] as String?,
      privacyNoticeVersion: map['privacy_notice_version'] as String?,
      evidenceTypes: List<Map<String, dynamic>>.from(
        map['evidence_types'] as List? ?? const [],
      ),
    );
  }

  final String? id;
  final String? role;
  final String status;
  final String? evidenceRoute;
  final int verificationLevel;
  final bool marketplaceEnabled;
  final bool guardianModeOptional;
  final String verificationMode;
  final bool productionVerified;
  final bool sandboxEligible;
  final bool testMode;
  final bool submissionsEnabled;
  final bool productionProviderAvailable;
  final bool canRetry;
  final bool supportEscalationAvailable;
  final bool documentsCollectedByMort;
  final String? environment;
  final String? provider;
  final String? providerReference;
  final String? decisionSource;
  final String? expiresAt;
  final String? rejectionCode;
  final String? publicMessage;
  final String? failureCode;
  final String? privacyNoticeVersion;
  final List<Map<String, dynamic>> evidenceTypes;
}

class SafetyIncidentSummary {
  const SafetyIncidentSummary({
    required this.id,
    required this.caseNumber,
    required this.category,
    required this.severity,
    required this.status,
    this.publicStatusNote,
    this.appealStatus = 'none',
  });

  factory SafetyIncidentSummary.fromMap(Map<String, dynamic> map) {
    return SafetyIncidentSummary(
      id: map['incident_id'] as String,
      caseNumber: map['case_number'] as String,
      category: map['category'] as String,
      severity: map['severity'] as String,
      status: map['status'] as String,
      publicStatusNote: map['public_status_note'] as String?,
      appealStatus: map['appeal_status'] as String? ?? 'none',
    );
  }

  final String id;
  final String caseNumber;
  final String category;
  final String severity;
  final String status;
  final String? publicStatusNote;
  final String appealStatus;
}

class SafetyCircleContact {
  const SafetyCircleContact({
    required this.id,
    required this.teenId,
    required this.relationshipLabel,
    required this.status,
    required this.permissions,
    this.contactId,
  });

  factory SafetyCircleContact.fromMap(Map<String, dynamic> map) {
    return SafetyCircleContact(
      id: map['id'] as String,
      teenId: map['teen_id'] as String,
      contactId: map['contact_id'] as String?,
      relationshipLabel: map['relationship_label'] as String,
      status: map['status'] as String,
      permissions: {
        'receive_safety_ping': map['receive_safety_ping'] == true,
        'receive_missed_checkin': map['receive_missed_checkin'] == true,
        'receive_job_summary': map['receive_job_summary'] == true,
        'receive_job_status': map['receive_job_status'] == true,
        'receive_emergency_request': map['receive_emergency_request'] == true,
        'view_limited_safety_plan': map['view_limited_safety_plan'] == true,
        'receive_completion': map['receive_completion'] == true,
      },
    );
  }

  final String id;
  final String teenId;
  final String? contactId;
  final String relationshipLabel;
  final String status;
  final Map<String, bool> permissions;
}

class JobSafetyAgreement {
  const JobSafetyAgreement({
    required this.id,
    required this.applicationId,
    required this.jobId,
    required this.teenId,
    required this.adultId,
    required this.version,
    required this.status,
    required this.terms,
  });

  factory JobSafetyAgreement.fromMap(Map<String, dynamic> map) {
    return JobSafetyAgreement(
      id: map['id'] as String,
      applicationId: map['application_id'] as String,
      jobId: map['job_id'] as String,
      teenId: map['teen_id'] as String,
      adultId: map['adult_id'] as String,
      version: (map['agreement_version'] as num).toInt(),
      status: map['status'] as String,
      terms: Map<String, dynamic>.from(map['terms_snapshot'] as Map),
    );
  }

  final String id;
  final String applicationId;
  final String jobId;
  final String teenId;
  final String adultId;
  final int version;
  final String status;
  final Map<String, dynamic> terms;
}

class AuthorizedLocationShare {
  const AuthorizedLocationShare({
    required this.id,
    required this.applicationId,
    required this.ownerId,
    required this.mode,
    required this.status,
    this.recipientUserId,
    this.coarseLocation,
    this.expiresAt,
  });

  factory AuthorizedLocationShare.fromMap(Map<String, dynamic> map) {
    return AuthorizedLocationShare(
      id: map['id'] as String,
      applicationId: map['application_id'] as String,
      ownerId: map['owner_id'] as String,
      recipientUserId: map['recipient_user_id'] as String?,
      mode: map['mode'] as String,
      status: map['status'] as String,
      coarseLocation: map['coarse_location'] as String?,
      expiresAt: map['expires_at'] as String?,
    );
  }

  final String id;
  final String applicationId;
  final String ownerId;
  final String? recipientUserId;
  final String mode;
  final String status;
  final String? coarseLocation;
  final String? expiresAt;
}

class AccountSessionSummary {
  const AccountSessionSummary({
    required this.reference,
    required this.userAgent,
    required this.assuranceLevel,
    required this.isCurrent,
  });

  factory AccountSessionSummary.fromMap(Map<String, dynamic> map) {
    return AccountSessionSummary(
      reference: map['session_reference'] as String,
      userAgent: map['user_agent'] as String? ?? 'Unknown device',
      assuranceLevel: map['assurance_level'] as String? ?? 'unknown',
      isCurrent: map['is_current'] == true,
    );
  }

  final String reference;
  final String userAgent;
  final String assuranceLevel;
  final bool isCurrent;
}

File: swift_mort/MORT/Models/TrustSafety.swift
import Foundation

enum IdentityEvidenceRoute: String, CaseIterable, Identifiable, Codable, Sendable {
    case schoolPhotoID = "school_photo_id"
    case governmentID = "government_id"
    case verifiedSchoolAccount = "verified_school_account"
    case approvedProgramID = "approved_program_id"
    case manualException = "manual_exception"

    var id: String { rawValue }
    var title: String {
        switch self {
        case .schoolPhotoID: "School photo ID"
        case .governmentID: "Government ID"
        case .verifiedSchoolAccount: "Verified school account"
        case .approvedProgramID: "Approved program ID"
        case .manualException: "Manual exception review"
        }
    }
}

enum IdentityEvidenceKind: String, CaseIterable, Identifiable, Codable, Sendable {
    case schoolPhotoID = "school_photo_id"
    case driversLicense = "drivers_license"
    case stateID = "state_id"
    case learnerPermit = "learner_permit"
    case passport = "passport"
    case passportCard = "passport_card"
    case otherGovernmentID = "other_government_id"
    case schoolAccountAssertion = "school_account_assertion"
    case accreditedProgramID = "accredited_program_id"
    case youthOrganizationID = "youth_organization_id"
    case homeschoolDocument = "homeschool_document"
    case supportingDocument = "supporting_document"
    case ownershipSelfie = "ownership_selfie"
    case addressDocument = "address_document"

    var id: String { rawValue }
    var title: String { rawValue.replacingOccurrences(of: "_", with: " ").capitalized }
}

struct IdentityEvidenceSummary: Codable, Hashable, Sendable {
    let type: String
    let status: String
    let submittedAt: String?

    enum CodingKeys: String, CodingKey {
        case type, status
        case submittedAt = "submitted_at"
    }
}

struct IdentityVerificationSummary: Codable, Hashable, Sendable {
    let ok: Bool
    let code: String?
    let id: UUID?
    let role: String?
    let status: String
    let evidenceRoute: String?
    let verificationLevel: Int
    let ageBand: String?
    let submittedAt: String?
    let reviewedAt: String?
    let expiresAt: String?
    let appealStatus: String?
    let rejectionCode: String?
    let marketplaceEnabled: Bool
    let guardianModeOptional: Bool
    let evidenceTypes: [IdentityEvidenceSummary]?
    let rawDocumentsVisibleToMarketplaceUsers: Bool?
    let verificationMode: String?
    let environment: String?
    let provider: String?
    let providerReference: String?
    let decisionSource: String?
    let verifiedAt: String?
    let productionVerified: Bool?
    let sandboxEligible: Bool?
    let testMode: Bool?
    let submissionsEnabled: Bool?
    let productionProviderAvailable: Bool?
    let publicMessage: String?

    enum CodingKeys: String, CodingKey {
        case ok, code, id, role, status
        case evidenceRoute = "evidence_route"
        case verificationLevel = "verification_level"
        case ageBand = "age_band"
        case submittedAt = "submitted_at"
        case reviewedAt = "reviewed_at"
        case expiresAt = "expires_at"
        case appealStatus = "appeal_status"
        case rejectionCode = "rejection_code"
        case marketplaceEnabled = "marketplace_enabled"
        case guardianModeOptional = "guardian_mode_optional"
        case evidenceTypes = "evidence_types"
        case rawDocumentsVisibleToMarketplaceUsers = "raw_documents_visible_to_marketplace_users"
        case verificationMode = "verification_mode"
        case environment, provider
        case providerReference = "provider_reference"
        case decisionSource = "decision_source"
        case verifiedAt = "verified_at"
        case productionVerified = "production_verified"
        case sandboxEligible = "sandbox_eligible"
        case testMode = "test_mode"
        case submissionsEnabled = "submissions_enabled"
        case productionProviderAvailable = "production_provider_available"
        case publicMessage = "public_message"
    }
}

struct TrustSafetyActionResult: Codable, Sendable {
    let ok: Bool
    let code: String?
    let message: String?
    let id: UUID?
    let status: String?
    let incidentID: UUID?
    let caseNumber: String?
    let agreementVersion: Int?
    let arrivalCode: String?
    let qrPayload: String?
    let expiresAt: String?
    let shareID: UUID?
    let marketplaceEnabled: Bool?
    let inviteCode: String?
    let environment: String?
    let provider: String?
    let providerReference: String?
    let testMode: Bool?
    let documentsAllowed: Bool?
    let productionEligible: Bool?

    enum CodingKeys: String, CodingKey {
        case ok, code, message, id, status
        case incidentID = "incident_id"
        case caseNumber = "case_number"
        case agreementVersion = "agreement_version"
        case arrivalCode = "arrival_code"
        case qrPayload = "qr_payload"
        case expiresAt = "expires_at"
        case shareID = "share_id"
        case marketplaceEnabled = "marketplace_enabled"
        case inviteCode = "invite_code"
        case environment, provider
        case providerReference = "provider_reference"
        case testMode = "test_mode"
        case documentsAllowed = "documents_allowed"
        case productionEligible = "production_eligible"
    }

    func requireSuccess(defaultMessage: String) throws -> TrustSafetyActionResult {
        guard ok else { throw MortError.backend(code: code ?? "unknown_permission_failure", message: message ?? defaultMessage) }
        return self
    }
}

struct IncidentCaseSummary: Codable, Hashable, Identifiable, Sendable {
    let incidentID: UUID
    let caseNumber: String
    let category: String
    let severity: String
    let status: String
    let publicStatusNote: String?
    let createdAt: String
    let updatedAt: String
    let appealStatus: String

    var id: UUID { incidentID }

    enum CodingKeys: String, CodingKey {
        case category, severity, status
        case incidentID = "incident_id"
        case caseNumber = "case_number"
        case publicStatusNote = "public_status_note"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case appealStatus = "appeal_status"
    }
}

struct SafetyCircleMember: Codable, Hashable, Identifiable, Sendable {
    let id: UUID
    let teenID: UUID
    let contactID: UUID?
    let relationshipLabel: String
    let status: String
    let receiveSafetyPing: Bool
    let receiveMissedCheckin: Bool
    let receiveJobSummary: Bool
    let receiveJobStatus: Bool
    let receiveEmergencyRequest: Bool
    let viewLimitedSafetyPlan: Bool
    let receiveCompletion: Bool

    enum CodingKeys: String, CodingKey {
        case id, status
        case teenID = "teen_id"
        case contactID = "contact_id"
        case relationshipLabel = "relationship_label"
        case receiveSafetyPing = "receive_safety_ping"
        case receiveMissedCheckin = "receive_missed_checkin"
        case receiveJobSummary = "receive_job_summary"
        case receiveJobStatus = "receive_job_status"
        case receiveEmergencyRequest = "receive_emergency_request"
        case viewLimitedSafetyPlan = "view_limited_safety_plan"
        case receiveCompletion = "receive_completion"
    }
}

struct JobSafetyAgreement: Codable, Hashable, Identifiable, Sendable {
    let id: UUID
    let applicationID: UUID
    let jobID: UUID
    let teenID: UUID
    let adultID: UUID
    let agreementVersion: Int
    let termsSnapshot: [String: JSONValue]
    let status: String
    let teenConfirmedAt: String?
    let adultConfirmedAt: String?

    enum CodingKeys: String, CodingKey {
        case id, status
        case applicationID = "application_id"
        case jobID = "job_id"
        case teenID = "teen_id"
        case adultID = "adult_id"
        case agreementVersion = "agreement_version"
        case termsSnapshot = "terms_snapshot"
        case teenConfirmedAt = "teen_confirmed_at"
        case adultConfirmedAt = "adult_confirmed_at"
    }
}

struct AuthorizedLocationShare: Codable, Hashable, Identifiable, Sendable {
    let id: UUID
    let applicationID: UUID
    let ownerID: UUID
    let recipientUserID: UUID?
    let mode: String
    let coarseLocation: String?
    let latitude: Double?
    let longitude: Double?
    let status: String
    let expiresAt: String
    let lastLocationAt: String?

    enum CodingKeys: String, CodingKey {
        case id, mode, latitude, longitude, status
        case applicationID = "application_id"
        case ownerID = "owner_id"
        case recipientUserID = "recipient_user_id"
        case coarseLocation = "coarse_location"
        case expiresAt = "expires_at"
        case lastLocationAt = "last_location_at"
    }
}

struct ActiveAccountSession: Codable, Hashable, Identifiable, Sendable {
    let sessionReference: String
    let createdAt: String?
    let updatedAt: String?
    let refreshedAt: String?
    let userAgent: String?
    let assuranceLevel: String?
    let isCurrent: Bool

    var id: String { sessionReference }

    enum CodingKeys: String, CodingKey {
        case sessionReference = "session_reference"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case refreshedAt = "refreshed_at"
        case userAgent = "user_agent"
        case assuranceLevel = "assurance_level"
        case isCurrent = "is_current"
    }
}
